hip-config.cmake.in

```diff
@@ -159,8 +159,27 @@ if(HIP_COMPILER STREQUAL "clang")
   if(NOT WIN32)
     find_dependency(AMDDeviceLibs)
   endif()
-  set(AMDGPU_TARGETS "gfx900;gfx906;gfx908;gfx90a;gfx1030" CACHE STRING "AMD GPU targets to compile for")
-  set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU targets to compile for")
+  set(GPU_DEFAULT_TARGETS "gfx900;gfx906;gfx908")
+  set(GPU_SUPPORTED_TARGETS "gfx701;gfx801;gfx802;gfx803;gfx900;gfx906;gfx908;gfx1010;gfx1011;gfx1012;gfx1030;gfx1031")
+  set(AMDGPU_TARGETS "${GPU_DEFAULT_TARGETS}" CACHE STRING "[Deprecated: please use GPU_TARGETS] AMD GPU targets to compile for. Changing this variable after cmake was called once will have NO effect!")
```

Comment:
I think it would be better to not define the variable, and then just check if its defined:

```cmake
if(NOT DEFINED AMDGPU_TARGETS)
    set(GPU_TARGETS "${GPU_DEFAULT_TARGETS}" CACHE STRING "GPU targets to compile for. Adding features is supported e.g. gfx908:xnack+")
else()
    message(WARNING "Variable AMDGPU_TARGETS is deprecated! Please use GPU_TARGETS to choose architectures.")
    set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU targets to compile for. Adding features is supported e.g. gfx908:xnack+")
endif()
```